lnschema_core/_core.py

```diff
@@ -497,10 +497,12 @@ def replace(self, data: Union[Path, str, pd.DataFrame, ad.AnnData], source: Opti
         self.hash = kwargs["hash"]
         self.suffix = kwargs["suffix"]
         self.source = kwargs["source"]
+        self.key = kwargs["key"]
```

Comment:
This is likely fine, but I hadn't yet thought through all possibilities.

Reply:
Yes, you're also right here. It's part of the user story of the other comment. I'll bring your changes back and add comments so that I don't forget about this.